internal/manager/run.go

```diff
@@ -162,6 +162,25 @@ func Run(ctx context.Context, c *Config, diagnostic util.ConfigDumpDiagnostic) e
 		return err
 	}
 
+	if featureGates[gatewayFeature] {
+		setupLog.Info("Gateway API Feature Enabled, Checking Existence of CRDs")
+		crdExistsCheckers := []crdExistsChecker{
+			gatewayCRDExistsCheceker,
+			gatewayClassCRDExistsChecker,
+			httpRouteCRDExistsChecker,
+		}
+		for _, checker := range crdExistsCheckers {
+			if !checker.CRDExists(mgr.GetClient()) {
+				setupLog.Info("CRD does not exist, disable gateway feature", "resource", checker.GVR.Resource)
+				featureGates[gatewayFeature] = false
+				// REVIEW: break here?
```

Comment:
Yep, break should be fine here. If any of the CRDs doesn't exist, we log that we can't use gateway and then we turn off the feature gate. That's adequate. :+1: 

```suggestion
```